### PR TITLE
MTM-59860 Microservice SDK updated to Spring Boot 3.3

### PR DESCRIPTION
**Pull request by kaczynskid-sag**
Caused by:
 - https://github.com/Cumulocity-IoT/cumulocity-clients-java-archived/pull/123

### CODE REVIEW

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Version **10.20.150.1** contains a fix for previously announced version **10.20.150.0** for applications configuring additional servlets. Please use this version when updating your application to Spring Boot 3.0. For details please refer to [Spring Boot 3.0 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#upgrade-to-spring-boot-3).
+
+Version **10.20.151.1** is based on Spring Boot 3.1 and is recommended as an intermediate step to limit the amount of changes done at once. For details please refer to [Spring Boot 3.1 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.1-Release-Notes).
+
+Version **10.20.153.1** is based on Spring Boot 3.2 and is another recommended intermediate step. For details please refer to [Spring Boot 3.2 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.2-Release-Notes).
+
+
+Finally, when upgrading to version **10.20.155.0** based on Spring Boot 3.3, the please refer to [Spring Boot 3.3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.3-Release-Notes).
```
Comment (jend-c8y) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md]:
```suggestion
Finally, when upgrading to version **10.20.155.0** based on Spring Boot 3.3, please refer to [Spring Boot 3.3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.3-Release-Notes).
```

**Review by jend-c8y — APPROVED** [review]
Please update suggested change

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  - value: change-inv-3bw8e
+    label: Announcement
+product_area: Application enablement & solutions
+component:
+  - value: component-Sv2buFZ5l
+    label: Microservice SDK
+build_artifact:
+  - value: tc-QHwMfWtBk7
+    label: cumulocity
+version: 10.20.155.0
```
Comment (mgrumann) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md:14]:
```suggestion
version: 1020.155.0
```

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ticket: MTM-58822
+---
+This is a follow-up on a recent [announcement](https://cumulocity.com/docs/change-logs/?change-type=.change-type-announcement%2C.change-type-api-change#cumulocity-undefined-microservices-sdk-spring-boot3-announcement) about migrating the Microservice SDK to Spring Boot 3.
+
+Starting from version **10.20.155.0**, the Microservice SDK is now using Spring Spring Boot 3.3.x, which marks the end of the recent dependency updates published in quick succession. Notice that along Spring Boot, most other dependencies were also updated to eliminate all security vulnerabilities those dependencies had.
```
Comment (mgrumann) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md:19]: 1020.155.0. (note that major version is 1020 but not 10)

This applies to a number of versions mentioned mentioned further down in the note

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    label: Announcement
+product_area: Application enablement & solutions
+component:
+  - value: component-Sv2buFZ5l
+    label: Microservice SDK
+build_artifact:
+  - value: tc-QHwMfWtBk7
+    label: cumulocity
+version: 10.20.155.0
+ticket: MTM-58822
```
Comment (mgrumann) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md]:
```suggestion
ticket: MTM-61583
```

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md (hunk 2 above)
Comment (kaczynskid-sag) [reply content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md:14]: All previous change logs are published with a dot separating the `10` from the next section. If we change it, then everywhere, but it's a separate discussions. For the moment it's consistent with what we have.

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md (hunk 3 above)
Comment (kaczynskid-sag) [reply content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md:19]: It's consistent with all other announcements we have - see above.

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+version: 10.20.155.0
+ticket: MTM-61583
+---
+This is a follow-up on a recent [announcement](https://cumulocity.com/docs/change-logs/?change-type=.change-type-announcement%2C.change-type-api-change#cumulocity-undefined-microservices-sdk-spring-boot3-announcement) about migrating the Microservice SDK to Spring Boot 3.
+
+Starting from version **10.20.155.0**, the Microservice SDK is now using Spring Spring Boot 3.3.x, which marks the end of the recent dependency updates published in quick succession. Notice that along Spring Boot, most other dependencies were also updated to eliminate all security vulnerabilities those dependencies had.
+
+Because of the amount of changes we also published intermediate versions to help with the upgrade process of all the applications using the SDK.
+
+Version **10.20.150.1** contains a fix for previously announced version **10.20.150.0** for applications configuring additional servlets. Please use this version when updating your application to Spring Boot 3.0. For details please refer to [Spring Boot 3.0 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#upgrade-to-spring-boot-3).
```
Comment (BeateRixen) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md]:
```suggestion
Version **10.20.150.1** contains a fix for the previously announced version **10.20.150.0** for applications configuring additional servlets. Use this version when updating your application to Spring Boot 3.0. For details refer to the [Spring Boot 3.0 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#upgrade-to-spring-boot-3).
```

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+Version **10.20.151.1** is based on Spring Boot 3.1 and is recommended as an intermediate step to limit the amount of changes done at once. For details please refer to [Spring Boot 3.1 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.1-Release-Notes).
```
Comment (BeateRixen) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md]:
```suggestion
Version **10.20.151.1** is based on Spring Boot 3.1 and is recommended as an intermediate step to limit the amount of changes done at once. For details refer to the [Spring Boot 3.1 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.1-Release-Notes).
```

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+Version **10.20.153.1** is based on Spring Boot 3.2 and is another recommended intermediate step. For details please refer to [Spring Boot 3.2 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.2-Release-Notes).
```
Comment (BeateRixen) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md]:
```suggestion
Version **10.20.153.1** is based on Spring Boot 3.2 and is another recommended intermediate step. For details refer to the [Spring Boot 3.2 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.2-Release-Notes).
```

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+
+Finally, when upgrading to version **10.20.155.0** based on Spring Boot 3.3, please refer to [Spring Boot 3.3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.3-Release-Notes).
```
Comment (BeateRixen) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md]:
```suggestion
Finally, when upgrading to version **10.20.155.0** based on Spring Boot 3.3, refer to the [Spring Boot 3.3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.3-Release-Notes).
```

**Review by BeateRixen — APPROVED** [review]

File: content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+version: 10.20.155.0
+ticket: MTM-61583
+---
+This is a follow-up on a recent [announcement](https://cumulocity.com/docs/change-logs/?change-type=.change-type-announcement%2C.change-type-api-change#cumulocity-undefined-microservices-sdk-spring-boot3-announcement) about migrating the Microservice SDK to Spring Boot 3.
+
+Starting from version **10.20.155.0**, the Microservice SDK is now using Spring Spring Boot 3.3.x, which marks the end of the recent dependency updates published in quick succession. Notice that along Spring Boot, most other dependencies were also updated to eliminate all security vulnerabilities those dependencies had.
+
+Because of the amount of changes we also published intermediate versions to help with the upgrade process of all the applications using the SDK.
+
+Version **10.20.150.1** contains a fix for the previously announced version **10.20.150.0** for applications configuring additional servlets. Use this version when updating your application to Spring Boot 3.0. For details refer to the [Spring Boot 3.0 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#upgrade-to-spring-boot-3).
```
Comment (mgrumann) [inline content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md:23]: Addtl. we need to mention that starting from 1020.150.1 there is a dependency on the server as mentioned in the communication with the US Sales team.

There is a dependency between client- and server-side. Tenants, which are configured to use the OAI-secure authentication method, will only function correctly with the latest maintenance release of the Cumulocity component on the server side. It needs to be at least core build 1020.651.0 for public environments, and core build 1018.540.258 for the y2024 release. There is no active check in the SDK to ensure this prerequisite is met.

Comment (kaczynskid-sag) [reply content/change-logs/application-enablement/cumulocity-1020-155-0-microservices-sdk-spring-boot-33.md:23]: As evident by internal communication, such a statement is confusing in this context. A separate change log on the Core side regarding a bug fix that may be made evident using certain versions of SDK would be more appropriate in this situation.

**Pull request merged**